droid-core/src/test/java/uk/gov/nationalarchives/droid/core/signature/compiler/ByteSequenceSerializerTest.java

```diff
@@ -103,12 +465,11 @@ private void testRenderSignatures(String fileName, SignatureType type) throws Ex
             if (sequence.trim().length() > 0) { //ignore blank lines in the test input files.
                 try {
                     compileExpression(sequence, type);
-                } catch (CompileException e) {
-                    failureMessages += e.getMessage();
-                    failed = true;
-                } catch (ParseException e) {
-                    failureMessages += e.getMessage();
+                } catch (CompileException | ParseException e) {
+                    failureMessages += (sequence + " : " + e.getMessage());
                     failed = true;
+                    compileExpression(sequence, type);
```

Comment:
For the test, once we have gathered that it failed, should we just move on to next expression, not sure about the purpose of `compileExpression` in the catch block

Reply:
The purpose of this test is to read all the signatures in the file (which is all of them that existed when the test was written), to ensure that we get no compilation exceptions when attempting to render them as different types of signature.

We gather all the failures together into a single message before failing the test, so we can see patterns in failures and not just play whack-a-mole as each individual one fails in turn.

Reply:
Sorry - I just realised what you were asking.

That should be taken out.  It was left in when I was debugging a failed expression - so I could put a breakpoint in after the failure.

Reply:
I've submitted a fix for this, removing the compileExpression() call.

Reply:
Thanks very much. 👍🏼